Fastfile.private

```diff
@@ -155,42 +157,65 @@ reflected in `Info.plist` during building."
   desc "It uses the same password stored in keychain for the current user."
   desc "Make sure to store safely the output of this command in the right `Google Drive` folder."
   private_lane :generate_push_certificates do |options|
-    build_configuration = options[:build_configuration]
+
+    build_configuration_key = options[:build_configuration]
+    build_configuration = get_build_configuration(build_configuration: build_configuration_key)
+    match_type = get_match_type(build_configuration: build_configuration_key)
+    is_development = match_type == Actions::GetMatchTypeAction::MATCH_TYPES[:test]
+    output_path = "#{build_configuration_key.to_s}_push_certificates"
 
     desc "Bundle identifier from xCode project"
     bundle_identifier = read_project_property(
-      build_configuration: get_build_configuration(build_configuration: build_configuration),
+      build_configuration: build_configuration,
       build_setting: 'PRODUCT_BUNDLE_IDENTIFIER'
     )
 
+    UI.message "Attempting to create push certificates for build configuration '#{build_configuration}'"
+    UI.message "Creating push certificates as: '#{is_development ? 'development' : 'production'}' and bundle ID: '#{bundle_identifier}'"
+    confirmation = UI.input "If the parameters are correct, proceed: Y/n"
+    unless confirmation.empty? || confirmation.downcase == "y"
+      UI.user_error! "Aborting due to parameters misconfiguration. Correct them and run the lane again."
+    end
+
     desc "Password for current user from keychain"
     user = CredentialsManager::AppfileConfig.try_fetch_value(:apple_id)
     keychain_entry = CredentialsManager::AccountManager.new(user: user)
     p12_password = keychain_entry.password
 
     pem(
       generate_p12: true,
-      development: get_match_type(build_configuration: build_configuration) == "development",
+      development: is_development,
       app_identifier: bundle_identifier,
       force: false,
       p12_password: p12_password,
-      output_path: "#{build_configuration.to_s}_push_certificates"
+      output_path: output_path
     )
+
   end
 
   desc "Builds the app creating the `.ipa` and `.dsym` files"
   private_lane :build_app do |options|
 
+    build_configuration_key = options[:build_configuration]
+    build_configuration = get_build_configuration(build_configuration: build_configuration_key)
+    match_type = get_match_type(build_configuration: build_configuration_key)
+
+    desc "Bundle identifier from xCode project"
+    bundle_identifier = read_project_property(
+      build_configuration: build_configuration,
+      build_setting: 'PRODUCT_BUNDLE_IDENTIFIER'
```

Comment:
I think we do this somewhere else. Maybe we can have an action that directly reads bundle identifier? Apart from the generic read property one.

Reply:
Yes, I agree, I've already seen this action being called with the same parameters.